include/itkCommandExhaustiveLog.h

```diff
@@ -0,0 +1,177 @@
+ * useful for obtaining data to plan future optimization attempts, such as whether
+ * a region appears rough or curved in a certain way that may render gradient descent
+ * less effective. Exhaustive data may also help inform the user in regards to
+ * the performance of ongoing optimization, such as overlaying optimizer steps
+ * onto an image of the exhaustive parametric region in order to determine whether
+ * the optimizer learning rate is appropriate for the region.
+ * 
+ * Wrapping for this class is limited to available wrappings for the internal
+ * itk::Image data array. Users wishing to make use of transforms with large
+ * parameter lists may find it necessary to build ITK with custom wrappings to
+ * meet their needs.
+ *
+ * Template parameters for class CommandExhaustiveLog:
+ *
+ * - TInternalData = Element type stored at each location in the data image.
+ * - TImageDimension = Dimension of image equal to number of transform parameters
+ *
+ * \ingroup ITKOptimizationMonitor
+ */
+template <typename TInternalData, unsigned int TImageDimension>
+class CommandExhaustiveLog : public itk::Command
+{
+public:
+  using Self = CommandExhaustiveLog;
+  using Superclass = itk::Command;
+  using Pointer = itk::SmartPointer<Self>;
+  itkNewMacro(Self);
+
+  /** Data type for pixel values in data image */
+  using InternalDataType = TInternalData;
+  /** Image dimension */
+  static constexpr unsigned int Dimension = TImageDimension;
+
+  // Only valid when used with ExhaustiveOptimizerv4 for now
+  // TODO allow different ExhaustiveOptimzerv4 templates
+  // TODO extend for other optimizers
+  using OptimizerType = itk::ExhaustiveOptimizerv4<double>;
+
+  /** Image used as n-dimensional array with positional aliasing. */
+  using ImageType = itk::Image<InternalDataType, Dimension>;
+  using ImagePointer = typename ImageType::Pointer;
+
+  /** Type to represent size of array along each dimension. */
+  using SizeType = typename ImageType::SizeType;
+  using SizeValueType = typename ImageType::SizeValueType;
+
+  /** Discrete values to access pixel location in data image. */
+  using IndexType = typename ImageType::IndexType;
+  /** Geometric distance between optimizer samples. */
+  using SpacingType = typename ImageType::SpacingType;
+  /** Geometric coordinates of a given sample index. */
+  using PointType = typename ImageType::PointType;
+
+  /** Default optimizer geometric output type. */
+  using ParametersType = typename OptimizerType::ParametersType;
+
+  /** Observe an event fired by calling object. */
+  void
+  Execute(itk::Object * caller, const itk::EventObject & event) override;
+
+  /** Observe an event fired by calling object. */
+  void
+  Execute(const itk::Object * caller, const itk::EventObject & event) override;
+
+  /** Retrieve a const reference to data point recorded during optimizer iteration. */
+  const TInternalData
+  GetData(const IndexType & index) const;
```

Comment:
Instead of `Data`, perhaps the more specific `Value`, to indicate the optimization value for these methods.